core/deployment/src/main/java/io/quarkus/deployment/configuration/BuildTimeConfigurationReader.java

```diff
@@ -267,7 +267,9 @@ ReadResult run() {
                 }
                 objectsByRootClass.put(clazz, instance);
                 String rootName = root.getRootName();
-                nameBuilder.append('.').append(rootName);
+                if (!rootName.isEmpty()) {
+                    nameBuilder.append('.').append(rootName);
```

Comment:
Actually I think that `.append(.)` shouldn't be there since `"quarkus."` is what is added by default. Am I right or am I missing something @dmlloyd ?

Reply:
Actually scratch that, I am making looking at wrong piece of code.

However I do see generate code containing:

```
        int var8 = var1.length();
        var1.append('.');
        var1.append("version");
        String var3 = var1.toString();
        Converter var4 = conv$0;
```

inside `initGroup$io$quarkus$runtime$TopLevelRootConfig` when I create a `TopLevelRootConfig`containing `version`

Reply:
I think https://github.com/quarkusio/quarkus/blob/bb3a6f47c56fc52598fdbc84fb18ee6ab2f96a00/core/deployment/src/main/java/io/quarkus/deployment/configuration/RunTimeConfigurationGenerator.java#L660 needs to be updated

Reply:
I have a quick and dirty fix for this.

Reply:
No, I don't think so, unless you can make a test that shows the problem?

Reply:
Sure yeah, I'll come up with one once I get back home.

Reply:
So here it is: https://github.com/geoand/quarkus/tree/pull/5989

Basically that branch is this PR plus one commit to add a `TopLevelRootConfig` (in https://github.com/geoand/quarkus/commit/4217332e6bad2e2a5156aa45ca6c02b6cbe44305).

If you use that branch and then just run the `main` integration test, everything fails.
When I debugged I found that the generated code was trying to load property `quarkus..profile` instead of `quarkus.profile` which of course doesn't exist.

Does that help?

Reply:
Yes! Thanks.

Reply:
Fix is pushed.

Reply:
Excellent, thanks!